elastic/kibana
# [Cases] Use templates when creating a case

## Summary

This PR adds the ability for users to select a template when creating a case. When a user selects a template the cases form fields are populated with the values from the template. The solution picker in the flyout changed as:

<img width="1281" alt="Screenshot 2024-06-20 at 1 26 14 PM" src="https://github.com/elastic/kibana/assets/7871006/30859c61-94e7-4f0a-97d1-cb081bd72c36">

## Bug fixes

1. Changing the solution through the solution picket in the flyout would not respect the connector's configuration.
2. Bug where the lazy loading of connectors' icons was causing the flyout to open and close when you selected a solution picket. This bug appeared when the first bug was fixed.

## Testing

1. Go to all solutions and create cases setting all fields (including all fields of all supported connectors) without using templates. Verify that everything is working as expected.
2. Go to all solutions, create different templates on each solution, and verify that when creating a case you can use templates and everything is working as expected.
3. Go to the alerts table of o11y and security and attach alerts to a new case. Verify that in the flyout the templates are working as expected.
4. Go to ML and try to attach an ML visualization to a new case. Verify that the solution picker is working as expected and it resets the form when changing solutions.

Fixes: https://github.com/elastic/kibana/issues/181871
Depends on: https://github.com/elastic/kibana/pull/185877

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
